File: src/main/kotlin/com/diving/wsql/temp/annotations/SqlQuery.java
package com.diving.wsql.temp.annotations;



import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;


@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface SqlQuery {
    //这个uk只有在做主类查询的时候才会有用
    String uk();
    String tableName();
    boolean distinct() default false;
}

File: src/main/kotlin/com/diving/wsql/temp/annotations/SqlQuery.kt
package com.diving.wsql.temp.annotations

import java.lang.annotation.*
import java.lang.annotation.Retention
import java.lang.annotation.Target

@Documented
@Target( ElementType.FIELD,ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
annotation class SqlQuery(val uk:String, val tableName: String,val distinct:Boolean=false)
